src/api/Elastic.Documentation.Api.Core/Search/SearchUsecase.cs

```diff
@@ -2,32 +2,37 @@
 // Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
 // See the LICENSE file in the project root for more information
 
+using Microsoft.Extensions.Logging;
+
 namespace Elastic.Documentation.Api.Core.Search;
 
-public class SearchUsecase(ISearchGateway searchGateway)
+public class SearchUsecase(ISearchGateway searchGateway, Logger<SearchUsecase> logger)
 {
 	public async Task<SearchResponse> Search(SearchRequest request, Cancel ctx = default)
 	{
-
-		// var validationResult = validator.Validate(request);
-		// if (!validationResult.IsValid)
-		// 	throw new ArgumentException(validationResult.Message);
-
 		var (totalHits, results) = await searchGateway.SearchAsync(
 			request.Query,
 			request.PageNumber,
 			request.PageSize,
 			ctx
 		);
 
-
-		return new SearchResponse
+		var response = new SearchResponse
 		{
 			Results = results,
 			TotalResults = totalHits,
 			PageNumber = request.PageNumber,
 			PageSize = request.PageSize,
 		};
+
+		logger.LogInformation("Search completed: {search.result.page_size} results for query '{search.query.text}' (Page {search.result.page_number}): {search.result.urls}",
```

Comment:
I am not sure if squeezing search.result.urls in the log message is going to be useful. 

We can make it a log attribute/property only see `[LogProperties]` when using the compile time log emitters.

https://andrewlock.net/customising-the-new-telemetry-logging-source-generator/#adding-properties-to-all-logs-using-enrichment